src/dataset/impl_dataset.rs

```diff
@@ -118,6 +121,15 @@ impl<R: Records, S> DatasetBase<R, S> {
 
         self
     }
+
+    /// Updates the target names of a dataset
+    pub fn with_target_names<I: Into<String>>(mut self, names: Vec<I>) -> DatasetBase<R, S> {
+        let target_names = names.into_iter().map(|x| x.into()).collect();
+
+        self.target_names = target_names;
+
+        self
+    }
```

Comment:
This method should check the length of the input vector so that it's equal to number of targets. The input should be Vec<String> so that it can be assigned directly. You can also try implementing this change for `feature_names`, but that might require more work to change.

Reply:
the input for the function is already **Vec(I)** where **I** implements **Into(String)**. Unless I am mistaken you want something like?

```
pub fn with_target_names(mut self, names: Vec<String>) -> DatasetBase<R, S> {
        if names.len() == self.ntargets() {
            self.target_names = names;
        } else {
            // raise some error to user stating the number of targets is X or default to class_{0..ntargets}?
        }
        self
    }
```

Reply:
Yeah pretty much. I don't mind panicking here, so you can just `assert` the condition.

Reply:
Gotcha